### PR TITLE
Scte35 2020 unit tests for 0x20,0x21, 0x22,0x23, 0x30, 0x31, 0x44 and 0x45

### PR DESCRIPTION
**Pull request by nehashanbhag**

### CODE REVIEW

File: scte35/state_test.go [hunk 1]
```diff
@@ -125,16 +125,19 @@ var scteDesc10Signal = "/DBFAAAAABeOAP/wBQb+WxdHNQAvAi1DVUVJACSqm3/9ABNP17wMGURJ
...
-// 0x23 BreakStart
+var scteDesc22Signal1  ="/DAsAAAAAAAAAP/wBQb+Uq7jaQAWAhRDVUVJAAABAH//AAGhbPAAACIAAL5jGRA="
+// 0x23 BreakEnd
 var scteDesc23Signal = "/DBAAAAAAtaWAAAABQb/e3rvuQAqAihDVUVJ/////3//AAEgZ5IBFGJyYXZvX0VQMDExMzQ0MTkwMTIyIwIEzcrAYA=="
 // 0x30 ProviderAdStart
-var scteDesc30Signal = "/DBIAAAAAtaWAAAABQb/elu5JQAyAjBDVUVJ/////3//AAApJfYPHHVybjpuYmN1bmkuY29tOmJyYzo1MjE1NDkxMTkwAgS9D8Yj"
+var scteDesc30Signal = "/DBPAAAAAAAAAP/wBQb/xHJKjwA5AAVTQVBTCwIwQ1VFSf////9//wAAKSd/Dxx1cm46bmJjdW5pLmNvbTpicmM6NTM5NjM4MjY2MAYCNMFiCw=="
+var scteDesc30Signal1 = "/DA2AAB6tBn0AP/wBQb//bDBCQAgAhRDVUVJYh5C/n+fAQVDMjQ3MDABAQAIQ1VFSQAAAACc3JW8"
 // 0x31 ProviderAdEnd
-var scteDesc31Signal = "/DBIAAAAAtaWAAAABQb/elu5JQAyAjBDVUVJ/////3//AAApJfYPHHVybjpuYmN1bmkuY29tOmJyYzo1MjE1NDkxMTkwAgS9D8Yj"
```
Comment (nehashanbhag) [inline scte35/state_test.go:137]: I had a 0x30 in place of 0x31 here by mistake earlier

File: scte35/state_test.go [hunk 2]
```diff
@@ -145,8 +148,9 @@ var scteDesc36Signal ="/DBLAAEs/S0UAP/wBQb+AAAAAAA1AjNDVUVJT///9X//AACky4AJH1NJR
 var scteDesc40Signal = "/DB7AAFfzZzVAP/wBQb+AAAAAABlAlJDVUVJAABeUX+XDUMJIUJMQUNLT1VUOjI1dU5ZeEl3UXVXclI5WUxDR2I0Y2c9PQ4eY29tY2FzdDpsaW5lYXI6bGljZW5zZXJvdGF0aW9uQAAAAg9DVUVJAABeUX+XAABBAAB1H+6A"
 // 0x44 ProviderAdBlockStart
 var scteDesc44Signal = "/DCJAAGqtdGtAP/wBQb/AAAAAQBzAnFDVUVJQAFfVH//AAApMuANXQ8edXJuOmNvbWNhc3Q6YWx0Y29uOmFkZHJlc3NhYmxlDyx1cm46bWVybGluOmxpbmVhcjpzdHJlYW06Mzg0OTU3MzQ4MjgzNzU2NTE2MwkNUE86MTA3MzgzMTc2NEQAAAAA5o/+Nw=="
+var scteDesc44Signal1 ="/DCfAAGab0HiAP/wBQb/AAAAAQCJAodDVUVJQAFfVH//AAApMuANcw8edXJuOmNvbWNhc3Q6YWx0Y29uOmFkZHJlc3NhYmxlDyx1cm46bWVybGluOmxpbmVhcjpzdHJlYW06Mzg0OTU3MzQ4MjgzNzU2NTE2MwkNUE86MTA3MzgzMTc2NAkKQlJFQUs6MTIzNAgIAAAAADeUEfJEAQMAAAu+e9A="
 // 0x45 ProviderAdBlockEnd
-var scteDesc45Signal = "/DCJAAGtnMd/AP/wBQb/AAAAAQBzAnFDVUVJQAFfVH//AAApMuANXQ8edXJuOmNvbWNhc3Q6YWx0Y29uOmFkZHJlc3NhYmxlDyx1cm46bWVybGluOmxpbmVhcjpzdHJlYW06Mzg0OTU3MzQ4MjgzNzU2NTE2MwkNUE86MTA3MzgzMTc2NEQAAAAAAy6WZA=="
```
Comment (nehashanbhag) [inline scte35/state_test.go:149]: This was a 0x44 instead of 0x45 by mistake

File: scte35/state_test.go [hunk 3]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
+
+	closed, err := state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:975]: `(have %d)`

File: scte35/state_test.go [hunk 4]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
```
Comment (laba2346) [inline scte35/state_test.go:1033]: `(have %d)`

File: scte35/state_test.go [hunk 5]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
```
Comment (laba2346) [inline scte35/state_test.go:1092]: `(have %d)`

File: scte35/state_test.go [hunk 6]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
```
Comment (laba2346) [inline scte35/state_test.go:1186]: `(have %d)`

File: scte35/state_test.go [hunk 7]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
```
Comment (laba2346) [inline scte35/state_test.go:1226]: `(have %d)`

File: scte35/state_test.go [hunk 8]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
+	}
+	if len(closed) != 4 {
+		t.Errorf("4 events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:1298]: `(have %d)`

File: scte35/state_test.go [hunk 9]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
+	}
+	if len(closed) != 1 {
+		t.Errorf("1 event should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:1316]: `(have %d)`

File: scte35/state_test.go [hunk 10]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
+
+	closed, err := state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:1338]: `(have %d)`

File: scte35/state_test.go [hunk 11]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
+	}
+	if len(closed) != 1 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 0 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
+	}
```
Comment (laba2346) [inline scte35/state_test.go:1357]:
```suggestion
	if len(state.Open()) != 0 {
		t.Errorf("There should be 0 open signals (have %d)", len(state.Open()))
	}
```

File: scte35/state_test.go [hunk 12]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
+
+	closed, err := state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
+	}
+	if len(closed) != 0 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:1378]: `(have %d)`

File: scte35/state_test.go [hunk 13]
```diff
@@ -941,4 +951,658 @@ func Test11Closing34_36_40_44(t *testing.T) {
...
+
+	closed, err = state.ProcessDescriptor(inSignal.Descriptors()[0])
+	if err != nil {
+		t.Errorf("ProcessDescriptor returned an error unexpectedly: %s", err.Error())
+	}
+	if len(closed) != 1 {
+		t.Errorf("1 event should have been closed (%d were)", len(closed))
+	}
+	if len(state.Open()) != 1 {
+		t.Errorf("There should be 1 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:1396]: `(have %d)`

**Review by laba2346 — APPROVED** [review]

**Pull request merged**